cypress/e2e/addProject.cy.js

```diff
@@ -0,0 +1,56 @@
+// testing suite for add project modal component
+
+describe('Add Project Modal Functionality', () => {
+    beforeEach(() => {
+        // login using custom command found in /support/commands.js
+        cy.login('kingliver', 'cats');
+    });
+
+    it('renders add project button', () => {
+        cy.get('[data-test="add-project-btn"]').should('be.visible')
+          .and('contain', 'Add Project');
+      });
+
+    it('renders the Add Project modal when the button is clicked', () => {
+        cy.get('[data-test="add-project-btn"]').click();
+        cy.get('[data-test="add-project-modal"]').should('be.visible');
+    });
+
+    it('adds a new project', () => {
+        const projectName = 'New Project';
+        // opens add project modal
+        cy.get('[data-test="add-project-btn"]').click();
+        // checks the modal is visible
+        cy.get('[data-test="add-project-modal"]').should('be.visible');
+        // types the project name into the input
+        cy.get('[data-test="add-project-name-input"]')
+            .type(projectName)
+            .should('have.value', projectName);
+        // submits the form
+        cy.get('[data-test="add-project-form"]').submit();
+        // verifies that the confirmation model appears and is correct
+        cy.get('[data-test="confirmation-modal"]').should('be.visible');
+        cy.get('[data-test="confirmation-modal"]').contains(`Project "${projectName}" has been added successfully!`);
+
+        // // if the user clicks home
+        // // closes the confirmation modal to go to home
+        // cy.get('[data-test="home-btn-confirmation-modal"]').click();
+        // // checks that the user is redirected to the home page
+        // cy.url().should('include', '/home');
+        // // checks if the project is displayed in the project list
+        // cy.get('[data-test="project-list"]').contains(projectName).should('exist');
+        
```

Comment:
I did it just to see it would pass without having to rewrite everything above twice, but I didn't think about it being a valuable functionality, which is a good point! 

I added a testing suite to test both routes and used the beforeEach to run the similar logic it takes to get to the individual routes. 

## Testing Screenshot:
<img width="467" alt="add-project-modal-testing" src="https://github.com/user-attachments/assets/b883bbf9-f126-43b6-a333-356f126de0a5">

Submitted changes with commit: [f26b6da](https://github.com/abbiehuynh/my-crochet-corner/pull/14/commits/f26b6daf1efdcdfe071d7fc3c6c3022103cc3791)